### DISCUSSION TITLE
How to migrate from multiple @repository to Definitions when repositories were loaded based on Environment

### DISCUSSION THREAD

**Discussion by jamiedemaria** [dagster-io/dagster#14292] [Q&A] [answered]
Surfacing from a Slack message:

I used to have multiple repositories loaded in the same User Code Docker Container.
The set of repositories loaded was configured through env vars when running my Docker Container.

Now that I have migrated to Definitions , it's not possible anymore.
* Is there a way to merge Definitions together to mitigate this?
* Is it possible to rename a JobDefinition, ScheduleDefinition or SensorDefinition , to add a prefix for example?

**Comment by jamiedemaria** [comment]
cc @schrockn

**Comment by schrockn** [comment, marked as answer]
You can still created named repositories using Definitions-style arguments using `create_repository_using_definitions_args`. If you want to retain the ability to create repositories you can use that API.

**Reply by ValetteValette** [reply]
Hello @schrockn,

Original poster on Slack here.

Very clear, thanks a lot for your answer 🙏🏻 

Is there any chance that the `Repository` API will be deprecated / removed in the near future?

**Reply by schrockn** [reply]
@ValetteValette no immediate plans. We are looking at new approaches to logically organizing definitions within a single code location that is more flexible and lighterweight than repositories. Until we have a solution that we are happy with there, we won't deprecate repositories as some users depend on them for this purpose.

**Reply by ValetteValette** [reply]
Ok, thanks a lot 🙏🏻
